Automattic/jetpack
# Various: Use wp_resource_hints

Closes #10352 ( props @willstocks )
Fixes #4838
Fixes #8090 ( This does not create a filter, but rather utilizes the existing WP filter that can be used for the ends requested in that issue. ).

When `Jetpack::dns_prefetch` was included, WordPress did not have a native solution, which was implemented awhile ago now.

#### Changes proposed in this Pull Request:
* Uses the Core provided functionality for `wp_resource_hints` instead of rolling out own.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Enable various modules (Comments, Comment Likes, Photon).
* View the `<head>` of a front-end page. Ensure the various Jetpack-related domains are present as `<link rel="dns-prefetch"...>`

#### Proposed changelog entry for your changes:
* Performance: Use WordPress-provided wp_resource_hints for DNS prefetching.
